frontend/web/components/ProjectUsage.tsx

```diff
@@ -0,0 +1,142 @@
+import { FC, useEffect, useState } from 'react'
+import Project from 'common/project'
+import InfoMessage from './InfoMessage'
+// import * as _data from 'common/data/base/_data'
+const _data = require('common/data/base/_data')
+
+type ProjectUsageType = {
+  projectId: string
+}
+
+type getProjectResponseType = {
+  max_features_allowed: number
+  max_segment_overrides_allowed: number
+  max_segments_allowed: number
+  total_features: 11
+  total_segments: 8
+  [key: string]: any
+}
+
+type usageType = {
+  max_features_allowed: number
+  max_segment_overrides_allowed: number
+  max_segments_allowed: number
+  total_features: number
+  total_segments: number
+  total_segment_overrides: {
+    [key: string]: number
+  }
+}
+
+type getEnvironmentsResType = {
+  results: {
+    api_key: string
+    [key: string]: any
+  }[]
+}
+
+type getEnvironmentResType = {
+  [key: string]: any
+  total_segment_overrides: number
+  id: number
+}
+
+const ProjectUsage: FC<ProjectUsageType> = ({ projectId }) => {
+  const [usage, setUsage] = useState<Partial<usageType>>({})
+
+  const getSegmentOverridesUsage = (apiKeys: string[]) => {
+    apiKeys.forEach((apiKey) => {
+      _data
+        .get(`${Project.api}environments/${apiKey}`)
+        .then((res: getEnvironmentResType) => {
+          setUsage((prev) => {
+            return {
+              ...prev,
+              total_segment_overrides: {
+                ...prev?.total_segment_overrides,
+                [apiKey]: res.total_segment_overrides,
+              },
+            }
+          })
+        })
+    })
+  }
+
+  const reduceTotalSegmentOverrides = () => {
+    if (!usage.total_segment_overrides) return 0
+    const total = Object.values(usage.total_segment_overrides).reduce(
+      (acc, curr) => acc + curr,
+      0,
+    )
+    return total
+  }
+
+  useEffect(() => {
+    _data
```

Comment:
This should use our CLI / standard redux approach,

Essentially if you do

```
npx ssg get
```

and follow the instructions, you can move getProjectResponseType to responses.ts and

Reply:
Oh, that's really cool.

It generated a new "useGetProjectQuery", that receives the projectId. The getProjectResponseType already existed there, it just wasn't being exported ai the Res object. So i deleted mine and put it on the Res Object.